opentelemetry-sdk/src/metrics/internal/histogram.rs

```diff
@@ -245,68 +209,25 @@ impl<T: Number> Histogram<T> {
     }
 }
 
-// TODO: uncomment once we know the reason for performance degradation
-// #[cfg(test)]
-// mod tests {
+#[cfg(test)]
+mod tests {
+    use super::*;
 
-//     use super::*;
-
-//     #[test]
-//     fn when_f64_is_nan_or_infinity_then_ignore() {
-//         struct Expected {
-//             min: f64,
-//             max: f64,
-//             sum: f64,
-//             count: u64,
-//         }
-//         impl Expected {
-//             fn new(min: f64, max: f64, sum: f64, count: u64) -> Self {
-//                 Expected {
-//                     min,
-//                     max,
-//                     sum,
-//                     count,
-//                 }
-//             }
-//         }
-//         struct TestCase {
-//             values: Vec<f64>,
-//             expected: Expected,
-//         }
-
-//         let test_cases = vec![
-//             TestCase {
-//                 values: vec![2.0, 4.0, 1.0],
-//                 expected: Expected::new(1.0, 4.0, 7.0, 3),
-//             },
-//             TestCase {
-//                 values: vec![2.0, 4.0, 1.0, f64::INFINITY],
-//                 expected: Expected::new(1.0, 4.0, 7.0, 3),
-//             },
-//             TestCase {
-//                 values: vec![2.0, 4.0, 1.0, -f64::INFINITY],
-//                 expected: Expected::new(1.0, 4.0, 7.0, 3),
-//             },
-//             TestCase {
-//                 values: vec![2.0, f64::NAN, 4.0, 1.0],
-//                 expected: Expected::new(1.0, 4.0, 7.0, 3),
-//             },
-//             TestCase {
-//                 values: vec![4.0, 4.0, 4.0, 2.0, 16.0, 1.0],
-//                 expected: Expected::new(1.0, 16.0, 31.0, 6),
-//             },
-//         ];
-
-//         for test in test_cases {
-//             let h = Histogram::new(vec![], true, true);
-//             for v in test.values {
-//                 h.measure(v, &[]);
-//             }
-//             let res = h.value_map.no_attribute_tracker.buckets.lock().unwrap();
-//             assert_eq!(test.expected.max, res.max);
-//             assert_eq!(test.expected.min, res.min);
-//             assert_eq!(test.expected.sum, res.total);
-//             assert_eq!(test.expected.count, res.count);
-//         }
-//     }
-// }
+    #[test]
+    fn check_buckets_are_selected_correctly() {
+        let hist = Histogram::<i64>::new(vec![1.0, 3.0, 6.0], false, false);
```

Comment:
We don't need this test.

We already have tests for checking `Histogram` with custom bounds: https://github.com/open-telemetry/opentelemetry-rust/blob/52f6d9214eca58c32e3fc41e3829038b868c07e1/opentelemetry-sdk/src/metrics/mod.rs#L319-L325

Reply:
Thanks for pointing that out.
would you mind if I move it from 'mod.rs' to 'histogram.rs' (this file) in this PR?

Reply:
> those are more like "e2e" tests or integration tests, and is best kept in mod.rs itself.

I agree with that too.

@fraillt Up to you if you want to update the existing test or keep this one here. I'm going to resolve this for now.